crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -514,48 +507,35 @@ impl AlgorithmUpdaterV1 {
 
     fn da_block_update(
         &mut self,
-        height_range: Range<u32>,
-        range_cost: u128,
+        heights: &[u32],
+        recording_cost: u128,
     ) -> Result<(), Error> {
-        let expected = self.da_recorded_block_height.saturating_add(1);
-        let first = height_range.start;
-        if first != expected {
-            Err(Error::SkippedDABlock {
-                expected,
-                got: first,
-            })
-        } else {
-            let last = height_range.end.saturating_sub(1);
-            let range_bytes = self.drain_l2_block_bytes_for_range(height_range)?;
-            let new_cost_per_byte: u128 = range_cost.checked_div(range_bytes).ok_or(
-                Error::CouldNotCalculateCostPerByte {
-                    bytes: range_bytes,
-                    cost: range_cost,
-                },
-            )?;
-            self.da_recorded_block_height = last;
-            let new_da_block_cost = self
-                .latest_known_total_da_cost_excess
-                .saturating_add(range_cost);
-            self.latest_known_total_da_cost_excess = new_da_block_cost;
-            self.latest_da_cost_per_byte = new_cost_per_byte;
-            Ok(())
-        }
+        let recorded_bytes = self.drain_l2_block_bytes_for_heights(heights)?;
+        let new_cost_per_byte: u128 = recording_cost.checked_div(recorded_bytes).ok_or(
+            Error::CouldNotCalculateCostPerByte {
+                bytes: recorded_bytes,
+                cost: recording_cost,
+            },
+        )?;
+        let new_da_block_cost = self
+            .latest_known_total_da_cost_excess
+            .saturating_add(recording_cost);
+        self.latest_known_total_da_cost_excess = new_da_block_cost;
+        self.latest_da_cost_per_byte = new_cost_per_byte;
+        Ok(())
     }
 
-    fn drain_l2_block_bytes_for_range(
+    fn drain_l2_block_bytes_for_heights(
         &mut self,
-        height_range: Range<u32>,
+        heights: &[u32],
     ) -> Result<u128, Error> {
         let mut total: u128 = 0;
-        for expected_height in height_range {
-            let (actual_height, bytes) = self
-                .unrecorded_blocks
-                .pop_first()
-                .ok_or(Error::L2BlockExpectedNotFound(expected_height))?;
-            if actual_height != expected_height {
-                return Err(Error::L2BlockExpectedNotFound(expected_height));
-            }
+        for expected_height in heights {
+            let bytes = self.unrecorded_blocks.remove(expected_height).ok_or(
+                Error::L2BlockExpectedNotFound {
+                    height: *expected_height,
+                },
+            )?;
```

Comment:
It's up to us. If this errors we have a serious problem with our code.

Reply:
If the committer has problems and submits several bundles per same height(they can in theory because they do re-bundle),  it will break the node.

Reply:
That can't happen if it's finalized before they report to us.

Reply:
I've been thinking about the (very unlikely) case that our gas price metadata is "ahead" of the chain on restart. This could in theory result in a bunch of the blocks being missing from the `unrecorded_blocks`. At that point we might have the da_committer reporting those blocks again. 

Since the precision of the algorithm isn't that important, I think the simplest solution would be to just ignore re-reported blocks.

In that case we don't want it throwing an error here. So I'm going to change this.

Reply:
In terms of the algorithm, I think the best approach here is to ignore the batch. The problem is if it ever occurs, we will carry those `unrecorded_blocks` forever. It's fine from the algorithm's perspective because the error will become negligible pretty quickly.

Reply:
Nevermind. We can both remove the blocks and ignore. I think that's the best.